ui/app/src/modules/Content/History/HistoryDialog.tsx

```diff
@@ -0,0 +1,583 @@
+            [menuOptions.compareTo, menuOptions.compareToCurrent, menuOptions.compareToPrevious],
+            [menuOptions.revertToThisVersion]
+          ],
+          anchorEl,
+          activeItem: version
+        });
+      }
+    },
+    [setMenu]
+  );
+
+  const handleItemClick = (version: LegacyVersion) => {
+    setCompareTo({ nextVersion: version });
+  };
+
+  const handleMenuClose = () => {
+    setMenu({
+      anchorEl: null,
+      activeItem: null
+    });
+  };
+
+  const handleMenuItemClicked = (section: SectionItem) => {
+    switch (section.id) {
+      case 'view': {
+        break;
+      }
+      case 'compareTo': {
+        setCompareTo({ version: menu.activeItem });
+        setMenu(menuInitialState);
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  const onPageChanged = (nextPage: number) => {
+    setPage(nextPage);
+  };
+
+  const handleBack = () => {
+    if (compareTo.nextVersion) {
+      setCompareTo({ nextVersion: null });
+    } else {
+      setCompareTo(compareToInitialState);
+    }
+  };
+
+  return (
+    <Dialog onClose={handleClose} open={open} fullWidth maxWidth="md">
+      <DialogHeader
+        title={
+          compareTo.version ?
+            (
+              compareTo.nextVersion ? (
+                <FormattedMessage
+                  id="historyDialog.comparingRevisions"
+                  defaultMessage={`Comparing Revisions -- {fileName}`}
+                  values={{ fileName: data.contentItem.internalName }}
+                />
+              ) : (
+                <FormattedMessage
+                  id="historyDialog.selectedRevisionToCompare"
+                  defaultMessage={`Select a revision to compare to "{version}"`}
+                  values={{ version: <FancyFormattedDate date={compareTo.version.lastModifiedDate}/> }}
+                />
+              )
+            ) : (
+              <FormattedMessage
+                id="historyDialog.headerTitle"
+                defaultMessage="Content Item History"
+              />
+            )
+        }
+        onClose={handleClose}
+        onBack={compareTo.version ? handleBack : null}
+      />
+      <DialogBody>
+        {
+          compareTo.version && compareTo.nextVersion ? (
+            <CompareRevision compareTo={compareTo}/>
+          ) : (
+            <SuspenseWithEmptyState
+              resource={resource}
+              component={HistoryList}
+              componentProps={{
+                handleOpenMenu,
+                handleItemClick,
+                rowsPerPage,
+                page,
+                compareTo
+              }}
+            />
+          )
+        }
+
+      </DialogBody>
+      {data.versions && (
+        <DialogFooter className={classes.dialogFooter}>
+          <TablePagination
+            className={classes.pagination}
+            classes={{ root: classes.pagination, selectRoot: 'hidden', toolbar: classes.toolbar }}
+            component="div"
```

Comment:
Seems unnecessary.

Reply:
which one?

Reply:
component= div is necessary, because we dont want a Table the default ones